src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -172,190 +185,132 @@ public bool ParseHeaders(TRequestHandler handler, ref SequenceReader<byte> reade
     {
         while (!reader.End)
         {
+            // Size of the detected EOL
+            var eolSize = -1;
+
+            // Check if the reader's span contains an LF to skip the reader if possible
             var span = reader.UnreadSpan;
-            while (span.Length > 0)
+
+            // Fast path, CR/LF at the beginning
+            if (span.Length >= 2 && span[0] == ByteCR && span[1] == ByteLF)
             {
-                byte ch1;
-                var ch2 = (byte)0;
-                var readAhead = 0;
+                reader.Advance(2);
+                handler.OnHeadersComplete(endStream: false);
+                return true;
+            }
 
-                // Fast path, we're still looking at the same span
-                if (span.Length >= 2)
-                {
-                    ch1 = span[0];
-                    ch2 = span[1];
-                }
-                else if (reader.TryRead(out ch1)) // Possibly split across spans
-                {
-                    // Note if we read ahead by 1 or 2 bytes
-                    readAhead = (reader.TryRead(out ch2)) ? 2 : 1;
-                }
+            var lfIndex = span.IndexOf(ByteLF);
```

Comment:
We should continue to fast-path a CR immediately followed by an LF with no CRs beforehand using `span.IndexOfAny(ByteCR, ByteLF)`. `IndexOfAny` not only avoids looping over the input twice, but it also uses SIMD operations to do a single comparison against each byte. We shouldn't scan every header twice for first for LF and then for RF.